feat(filters): implement create filter functionality

alonp99 opened the pull request:
- Add an API endpoint for creating filters
- Include UI enhancements for filter creation dialog

(Your filter creation process is so smooth, it's practically butter on a hot knife)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced workflows table with improved cell content, including copy ID functionality and styled displays.
  - Introduced filter creation functionality with a new dialog for user input.
  - Added vendor details display in the Workflow Definition component.
  - Updated UI Definition editor to support new viewing modes and structured layouts.
  - Improved debug mode display in the Collection Flow component with structured information and tooltips.

- **Bug Fixes**
  - Improved error handling in various components to enhance robustness.

- **Documentation**
  - Updated documentation to reflect changes in method signatures and new parameters.

- **Style**
  - Enhanced visual styling across multiple components for better user experience, including navigation buttons and debug information.

- **Tests**
  - Added tests to ensure new features function as expected.

- **Chores**
  - Cleaned up code and removed commented-out directives for better readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

alonp99 committed:
feat(filters): implement create filter functionality

- Add an API endpoint for creating filters
- Include UI enhancements for filter creation dialog

(Your filter creation process is so smooth, it's practically butter on a hot knife)

alonp99 committed:
fix(UIDefinitionEditor): resolve rendering issues in UI definition editor

- Ensure components correctly return when encountering undefined types
- Clean up nested element rendering logic for clarity

(if your nested elements were any deeper, they'd need their own zip code)

Omri-Levy commented on packages/common/src/schemas/documents/schemas/entity-schema.ts:
```diff
@@ -70,7 +70,91 @@ export const BusinessDataSchema = Type.Object({
   {
     type: Type.String({ enum: ['individual', 'business'] }),
-    data: Type.Union([IndividualDataSchema, BusinessDataSchema]),
+    data: Type.Union([
```
This change reverts mine like coderabbit mentioned.

Omri-Levy commented on packages/workflow-core/src/lib/plugins/common-plugin/workflow-token-plugin.ts:
```diff
@@ -19,7 +19,7 @@ export class WorkflowTokenPlugin {
     this.action = pluginParams.action;
     this.successAction = pluginParams.successAction;
     this.errorAction = pluginParams.errorAction;
-    this.persistResponseDestination = 'metadata';
```
Does an empty string write to the root of the object?

alonp99 committed:
feat(ui): enhance debugging and UI definition features

- Add a visual indication for debug mode in the CollectionFlow component
- Introduce theme and locales tabs in the UIDefinitionEditor
- Utilize dayjs for date formatting in the WorkflowDefinitionSummaryCard

(Your code looks nice, but I bet even your cat would prefer a more straightforward UI)

alonp99 committed:
feat(tables): enhance clipboard copy function with toast notifications

- Integrate toast notifications for clipboard copy success and failure
- Apply changes across multiple table column components

(your clipboard functionality needed a better alert system than just wishing on a star)

alonp99 committed:
chore(release): bump version to 0.7.80 and 0.6.76 for various packages

- Update version for backoffice-v2, kyb-app, and workflows-service to 0.7.80
- Upgrade workflow-browser-sdk and workflow-node-sdk to version 0.6.76

(your dependencies have more updates than my social media feed during a trend)

alonp99 committed:
refactor(schema): simplify data structure in entity schema

- Replace detailed inline object definitions with separate schemas for individuals and businesses
- Enhance readability and maintainability by utilizing existing schemas

(Your code reduction skills are sharp, just like the corners of the new schema)

alonp99 committed:
feat(workflow): enhance WorkflowDefinition with integration catalog

- Introduce a Dialog component for viewing the integration catalog
- Implement state management for opening and closing the dialog
- Add copy-to-clipboard functionality for vendor keys

(Your integration catalog is so fancy, it makes a five-star restaurant look like a food truck)

alonp99 committed:
chore: bump versions and update dependencies across multiple packages

- Update version numbers in various packages to their latest releases
- Upgrade dependent packages to maintain compatibility

(your projects are like a fine wine, they really should age before a release)
